src/main/java/org/gridsuite/explore/server/services/ExploreService.java

```diff
@@ -143,42 +151,35 @@ public ExploreService(
 
     public Mono<Void> deleteElement(UUID id, String userId) {
         return directoryService.getElementInfos(id).flatMap(elementAttributes -> {
-            if (elementAttributes.getType().equals(STUDY)) {
-                studyService.deleteFromStudyServer(elementAttributes.getElementUuid(), userId).subscribe();
-            } else if (elementAttributes.getType().equals(CONTINGENCY_LIST)) {
-                contingencyListService.deleteContingencyList(elementAttributes.getElementUuid()).subscribe();
-            } else if (elementAttributes.getType().equals(FILTER)) {
-                filterService.deleteFilter(elementAttributes.getElementUuid()).subscribe();
+            IDirectoryElementsService service = genericServices.get(elementAttributes.getType());
+            if (service == null) {
+                return Mono.error(new ExploreException(UNKNOWN_ELEMENT_TYPE, "Unknown element type :" + elementAttributes.getType()));
             }
-            return directoryService.deleteElement(id, userId);
+            return service.delete(id, userId).doOnSuccess(e -> directoryService.deleteElement(id, userId));
```

Comment:
return service.delete(id, userId).doOnSuccess(e -> directoryService.deleteElement(id, userId).**subscribe()**); ?
The directory delete isn't executed

Reply:
a subscribe is missing...

Reply:
done